packages/edit-site/src/components/global-styles/variations-panel.js

```diff
@@ -22,8 +32,10 @@ export function useBlockVariations( name ) {
 		},
 		[ name ]
 	);
-	const coreBlockStyles = getCoreBlockStyles( blockStyles );
-	return coreBlockStyles;
+	const [ variations ] = useGlobalStyle( 'variations', name, 'base' );
```

Comment:
At this point, the theme.json resolver has absorbed block style variations registered through the block styles registry into the merged global styles data so it can be used as a source of validated variations.